os/arch/arm/include/imxrt/chip.h

```diff
@@ -0,0 +1,157 @@
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ************************************************************************************/
+
+#ifndef __ARCH_ARM_INCLUDE_IMXRT_CHIP_H
+#define __ARCH_ARM_INCLUDE_IMXRT_CHIP_H
+
+/************************************************************************************
+ * Included Files
+ ************************************************************************************/
+
+#include <tinyara/config.h>
+
+/************************************************************************************
+ * Pre-processor Definitions
+ ************************************************************************************/
+
+/* Get customizations for each supported chip */
+
+#if defined(CONFIG_ARCH_CHIP_MIMXRT1051DVL6A) || \
+	 defined(CONFIG_ARCH_CHIP_MIMXRT1051CVL5A) || \
+	 defined(CONFIG_ARCH_CHIP_MIMXRT1052DVL6A) || \
+	 defined(CONFIG_ARCH_CHIP_MIMXRT1052CVL5A)
+/* MIMXRT1051CVL5A - Industrial, Reduced Features, 528MHz
+ * MIMXRT1051DVL6A - Consumer, Reduced Features, 600MHz
+ * MIMXRT1052CVL5A - Industrial, Full Feature, 528MHz
+ * MIMXRT1052DVL6A - Consumer, Full Feature, 600MHz
+ */
+
+#define IMXRT_OCRAM_SIZE            (512 * 1024)	/* 512Kb OCRAM */
+#define IMXRT_GPIO_NPORTS            5	/* Five total ports */
+
+#else
+#error "Unknown i.MX RT chip type"
+#endif
+
+/* NVIC priority levels *************************************************************/
+/* Each priority field holds an 8-bit priority value, 0-15. The lower the value, the
+ * greater the priority of the corresponding interrupt.  The i.MX RT processor
+ * implements only  bits[7:4] of each field, bits[3:0] read as zero and ignore writes.
+ */
+
+#define NVIC_SYSH_PRIORITY_MIN        0xf0	/* All bits[7:4] set is minimum priority */
+#define NVIC_SYSH_PRIORITY_DEFAULT    0x80	/* Midpoint is the default */
+#define NVIC_SYSH_PRIORITY_MAX        0x00	/* Zero is maximum priority */
+#define NVIC_SYSH_PRIORITY_STEP       0x40	/* Two bits of interrupt priority used */
+
+/* If CONFIG_ARMV7M_USEBASEPRI is selected, then interrupts will be disabled
```

Comment:
I don't think this explanation is required as you have removed the config ARMV7M_USEBASEPRI

Reply:
I cannot understand your comment.
do you mean that I should remove CONFIG_ARMV7M_USEBASEPRI definition?